create-k8s-chained-sessions.sh

```diff
@@ -6,63 +6,92 @@
+
+        # use the parent session to get the role arn
+        # so we don't have to hard-code account ids
+        leapp session start --sessionId "$parent_session_id" > /dev/null 2> >(logStdErr)
+        role_arn=$(aws iam get-role --role-name "$chained_role_name" --query Role.Arn | tr -d '"')
+        leapp session stop --sessionId "$parent_session_id" > /dev/null 2> >(logStdErr)
 
-    # create a named profile per account so they can be used simultaneously
-    echo "creating new profile"
-    createLeappProfile $parent_session_name
+        green_echo "    creating new profile"
+        profile_id=$(createLeappProfile "$parent_session_name")
 
-    echo "creating new session"
-    # create new chained leapp session from parent
-    leapp session add --providerType aws --sessionType awsIamRoleChained \
-        --sessionName $chained_session_name --region $REGION \
-        --roleArn $role_arn --parentSessionId $parent_session_id \
-        --profileId $PROFILE_ID
-    # add session id from the new session to CHAINED_SESSION_IDS
-    chained_session_id=$(leapp session list --columns=ID --filter="Session Name=${chained_session_name}" --no-header)
-    CHAINED_SESSION_IDS="${CHAINED_SESSION_IDS}${chained_session_name},${chained_session_id}\n"
+        green_echo "    creating new session"
+        leapp session add --providerType aws --sessionType awsIamRoleChained \
+            --sessionName "$chained_session_name" --region "$REGION" \
+            --roleArn "$role_arn" --parentSessionId "$parent_session_id" \
+            --profileId "$profile_id" > /dev/null 2> >(logStdErr)
+
+    else
+        yellow_echo "    existing session found"
+    fi
+}
+
+# @return the Leapp session ID of the session whose name is the first argument
+#   to this function, if one exists.
+function leappSessionId {
+    # The ^ and $ in the session filter are regex anchors to ensure we don't
+    # match e.g. both `chained-from-panorama-k8s-playground` and
+    # `chained-from-panorama-k8s-playground-2`.
+    leapp session list -x --filter="Session Name=^${1}$" --output json | jq -r ".[] | select(.role==\"${2}\") | .id"
 }
 
 # function to create a leapp profile to associate with the chained k8s sessions
 # stores the new profile id in PROFILE_ID
 function createLeappProfile {
+    # The ^ and $ in the session filter are regex anchors to ensure we don't
+    # match e.g. both `kubectl-access-role-panorama-k8s-playground` and
+    # `kubectl-access-role-panorama-k8s-playground-2`.
     profile_name="kubectl-access-role-${1}"
-    leapp profile create --profileName $profile_name
-    PROFILE_ID=$(leapp profile list --columns=ID --filter="Profile Name=${profile_name}" --no-header)
+    profile_id=$(leapp profile list -x --output json --filter="Profile Name=^${profile_name}$" | jq -r '.[].id')
+    if [[ -n "${profile_id}" ]]; then
+        echo "${profile_id}"
```

Comment:
Does this just output a UUID? I'm wondering if it would be clearer if it was something like:
```sh
echo "Profile ${profile_name} already exists with id: ${profile_id}"
```

Reply:
It doesn't actually output anything, echo is used here to return the value of the existing profile id from the function so we can use it in the session creation.